bin/ansible2gdeploy.py

```diff
@@ -28,6 +32,50 @@
 GLUSTER_SERVER_GROUP = "gluster-servers"
 GLUSTER_CLIENT_GROUP = "usm_client"
 
+
+def host_fqdn(fqdn):
+    """
+    Return host's fqdn.
+    """
+    print("FQDN: %s" % fqdn, file=sys.stderr)
+    return fqdn
+
+def host_short(fqdn):
+    """
+    Return host's short hostname.
+    """
+    short = fqdn.split(".")[0]
+    print("SHORT: %s" % short, file=sys.stderr)
+    return short
+
+def host_ip(fqdn):
+    """
+    Return host's IP.
+    """
+    ip = socket.gethostbyname(fqdn)
+    print("IP: %s" % ip, file=sys.stderr)
+    return ip
+
+def host_mixed(fqdn):
+    """
+    Return host's fqdn, hostname or IP - sequentially selected.
+    """
+    choices = (host_fqdn, host_short, host_ip)
+    try:
+        current_i = host_mixed.i
+    except AttributeError:
+        host_mixed.i = 0
+        current_i = 0
+    host_mixed.i = (host_mixed.i + 1) % len(choices)
+    return choices[current_i](fqdn)
+
+def host_random(fqdn):
+    """
+    Return host's fqdn, hostname or IP - randomly selected.
+    """
+    choices = (host_fqdn, host_short, host_ip)
+    return random.choice(choices)(fqdn)
```

Comment:
This random choice seems nice, but is it really useful when there is no guarantee that each host representation (ip address, short host name and fqdn) is used at least once? That said, I would keep it. I'm just saying I would not use this in automated test setup.

Reply:
@mbukatov I agree, the `mixed` version is more predictable and definitely better for regular automated test setup. But the `random` function was quite simple, so I put it here... it might be useful sometime.

Reply:
Agreed.